### PR TITLE
fix(VDataTable, VSimpleTable): missing rtl styles, other minor tweaks

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
## Description
- `.v-data-table__empty-wrapper` has hover effect again, regressed in 13816f144f9fe493df0dbfd3258d4258e91bb105
- added `pointer-events: none` to `.v-data-table__empty-wrapper`
- added missing rtl styles
- moved styles not related to theming outside of `theme()`

## How Has This Been Tested?
visually

## Markup:

<details>

```vue
<template>
  <v-container grid-list-xl>
    <v-btn block @click="$vuetify.rtl = !$vuetify.rtl" class="mb-12">toggle rtl: {{ $vuetify.rtl }}</v-btn>

    <v-data-table item-key="name" class="elevation-1 mb-12" loading loading-text="Loading... Please wait"></v-data-table>
  
    <v-data-table
      :headers="headers"
      :items="desserts"
      height="300"
      fixed-header
      class="elevation-1 mb-12"
    />

    <v-simple-table>
    <thead><tr><th>Foo</th><th>Bar</th></tr></thead>
    </v-simple-table>
  </v-container>
</template>

<script>
export default {
  data: () => ({
    headers: [
      { text: 'Dessert (100g serving)',  value: 'name' },
      { text: 'Calories', value: 'calories' },
      { text: 'Fat (g)', value: 'fat' },
    ],
    desserts: Array(50).fill({
      name: 'Frozen Yogurt',
      calories: 159,
      fat: 6.0,
    })
  })
}
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VDataTable/VSimpleTable.sass [hunk 1]
```diff
@@ -95,7 +100,11 @@
 
   // Account for scroll bar
   .v-table__header, .v-data-footer
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VDataTable/VSimpleTable.sass]: @nekosaur `v-table__header` exists only here - should it be removed or it's a typo in a class name?

Comment (nekosaur) [reply packages/vuetify/src/components/VDataTable/VSimpleTable.sass]: It is most likely a remnant of an earlier iteration of the refactor (specifically another virtual rows implementation I think). So should be safe to remove.

**Comment by johnleider** [comment]
Missing spacing between text and select in RTL

image

image

**Comment by jacekkarczmarczyk** [comment]
@johnleider updated

**Pull request merged**